composer/trainer/trainer.py

```diff
@@ -836,15 +876,30 @@ def __init__(
         # DDP.
 
         self._rng_state = None
+        # First try to load from ``load_path``.
         if load_path is not None:
+            # If load_ignore_missing_checkpoint is False, we emit a warning and continue if the checkpoint is
+            # missing. This is required for graceful resumption support, where the "latest" checkpoint
+            # will not exist at start of training.
             self._rng_state = load_checkpoint(state=self.state,
                                               path=load_path,
                                               object_store=load_object_store,
-                                              load_weights_only=load_weights_only,
+                                              load_weights_only=False,
                                               strict_model_weights=load_strict,
+                                              load_ignore_missing_checkpoint=load_ignore_missing_checkpoint,
                                               chunk_size=load_chunk_size,
                                               progress_bar=load_progress_bar)
             reproducibility.seed_all(self.state.seed)
+        # If ``load_path`` is not specified or failed, next try ``load_only_weights_path``.
+        if load_only_weights_path is not None and self._rng_state is None:
```

Comment:
Would it work to catch catch the `FileNotFound` exception directly to make this clearer of when we will use the initial weights path. (Conditioning this on self.)rng_state is None is a bit tricky of why that is the case)